### PR TITLE
[twitter] Add probe_detect hook function

### PR DESCRIPTION
**Pull request by MrPetovan**
Adds Twitter profile probe via the API.

Requires https://github.com/friendica/friendica/pull/8741

### CODE REVIEW

File: twitter/twitter.php [hunk 1]
```diff
@@ -1063,34 +1071,52 @@ function twitter_get_relation($uid, $target, $contact = [])
...
+		'url'      => $url,
+		'network'  => Protocol::TWITTER,
+		'alias'    => 'twitter::' . $data->id_str,
+		'baseurl'  => $baseurl,
+		'name'     => $data->name,
+		'nick'     => $data->screen_name,
+		'addr'     => $addr,
+		'location' => $data->location,
+		'about'    => $data->description,
+		'photo'    => $avatar,
```
Comment (annando) [inline twitter/twitter.php]: In the contact table the field "avatar" is meant to store the real path to the avatar picture. In "photo" there is the cropped and cached version.

File: twitter/twitter.php [hunk 2]
```diff
@@ -1063,34 +1071,52 @@ function twitter_get_relation($uid, $target, $contact = [])
...
 		$cid = $pcontact['id'];
 	} else {
-		$cid = Contact::getIdForURL($url, 0, true, $fields);
+		$cid = Contact::getIdForURL($fields['url'], 0, true, $fields);
 	}
 
 	if (!empty($cid)) {
 		DBA::update('contact', $fields, ['id' => $cid]);
-		Contact::updateAvatar($avatar, 0, $cid);
+		Contact::updateAvatar($fields['photo'], 0, $cid);
```
Comment (annando) [inline twitter/twitter.php]: The function `updateAvatar` only stores the different versions of the picture (in "photo", "thumb", "micro") when the provided url differs from the existing value in the "avatar" field. Means: We mustn't store picture data with the `DBA::update` the line before. This had to be done solely with `Contact::updateAvatar`.

File: twitter/twitter.php [hunk 3]
```diff
@@ -1164,48 +1190,31 @@ function twitter_fetch_contact($uid, $data, $create_user)
...
-
-	if ($user_id != "") {
-		$parameters["user_id"] = $user_id;
-	}
-
-	// Fetching user data
-	$connection = new TwitterOAuth($ckey, $csecret, $otoken, $osecret);
 	try {
+		// Fetching user data
+		$connection = new TwitterOAuth($ckey, $csecret);
```
Comment (annando) [inline twitter/twitter.php:1208]: We don't need `$otoken` and `$osecret` here?

**Comment by annando** [comment]
Hmm, I'm very nitpicking today, I guess.

File: twitter/twitter.php (hunk 1 above)
Comment (MrPetovan) [reply twitter/twitter.php]: Got it.

File: twitter/twitter.php (hunk 3 above)
Comment (MrPetovan) [reply twitter/twitter.php:1208]: Nope, because we don't retrieve data specific to a user.

Comment (annando) [reply twitter/twitter.php:1208]: When its working then I'm fine with that.

**Comment by MrPetovan** [comment]
I like you that way. ❤

File: twitter/twitter.php (hunk 2 above)
Comment (MrPetovan) [reply twitter/twitter.php]: Done.

**Pull request merged**